libraries/abstractions/common_io/test/test_iot_spi.c

```diff
@@ -1068,35 +944,117 @@ TEST( TEST_IOT_SPI, AFQP_IotSPI_TransferAsyncAssisted )
         TEST_ASSERT_EQUAL( pdTRUE, lRetVal );
     }
 
-    /* restore original configuration */
+    /* Restore the original configuration saved in the beginning of this test,
+     * in order to reset to the original state before this test. */
     lRetVal = iot_spi_ioctl( xSPIHandle, eSPISetMasterConfig, &xOrigConfig );
     TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
 
-    /* Close SPI handle */
     lRetVal = iot_spi_close( xSPIHandle );
     TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
 
-    /* restore slave select */
+    /* Restore the original selected slave device, in order to reset to the original state before this test. */
+    if( ulAssistedTestIotSpiSlave != ultestIotSpiSlave )
+    {
+        #ifdef IOT_TEST_COMMON_IO_SPI_SLAVE_SELECT_SUPPORTED
```

Comment:
Should the #define instead surround this whole block?

Reply:
That makes sense.

Reply:
All such "slave select" coded is in assisted tests. I suggest let's not worry about too much about assisted tests because they are not supported/called yet anyway, even they are included in the test file compile.

Reply:
It doesn't look to me like this code meets a high bar in its current form. Can we keep it out of our repo until we need it and it makes more sense to invest the effort?